### PR TITLE
MDEV-9158 Read the file in chunks and stop to read if it has read more than MAX_KEY_FILE_SIZE

### PR DESCRIPTION
**Pull request by dingweiqings**
- [x] *The Jira issue number for this PR is: MDEV-9158*

## Description
1. I change the key file max size check implemention. Now file key plugin reads  max key file size at most. For this way , file key plugin can support read from an named pipe file in linux.
2. A test for too large file key 
3. A test for named pipe file key

## How can this PR be tested?
As usual , make CI test.

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [x] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

## Backward compatibility
No

### CODE REVIEW

**Comment by dingweiqings** [comment]
I have a question . In plugin/file_key_management/parser.cc ,I want to print how many bytes has been read .But I don't know use which print fuction after I search some keyword in KB

**Comment by dingweiqings** [comment]
I run ```./mtr --suite=encryption ``` and all tests are passed
image

File: plugin/file_key_management/parser.cc [hunk 1]
```diff
@@ -335,7 +335,7 @@ char* Parser::read_and_decrypt_file(const char *secret)
                    ME_ERROR_LOG , filename, errno);
     goto err2;
   }
-  fprintf(stderr, "Read from %s , filesize: %ldB,max key file size :%d B\n " , filename,(size_t)file_size,MAX_KEY_FILE_SIZE);
+  fprintf(stderr, "Read from %s , filesize: %ldB,max key file size :%d B\n " , filename,(ulong)file_size,MAX_KEY_FILE_SIZE);
```
Comment (grooverdan) [inline plugin/file_key_management/parser.cc:338]: note a `ssize_t`  uses format `%zd` and then won't need a cast regardless of platform.

Comment (dingweiqings) [reply plugin/file_key_management/parser.cc:338]: Thanks a lot . And here I use ```fprintf`` .Is there other better log function instead?

Comment (grooverdan) [reply plugin/file_key_management/parser.cc]: `my_printf_error` seems the best I could find.

Comment (dingweiqings) [reply plugin/file_key_management/parser.cc]: Let me think, But this maybe is not an error . This real read bytes is logged to tell user and diagnose when user provide a large key file.

Comment (dingweiqings) [reply plugin/file_key_management/parser.cc]: Any comments would be great on codes .

Comment (dingweiqings) [reply plugin/file_key_management/parser.cc]: > `my_printf_error` seems the best I could find.

Haha, I understand what you mean and it works . Error log file  is maraidb  log file by design.

**Comment by dingweiqings** [comment]
@grooverdan Any comments is great for this PR.

**Comment by grooverdan** [comment]
Sorry, have been a bit busy. This will need rebasing once #2269 is merged up and might need a allocation of `MAX_SECRET_SIZE + 1` depending on how you rework the implementation.

**Comment by dingweiqings** [comment]
use #2297 instead , so close it .

**Pull request closed without merging**